system/Commands/Database/ShowTableInfo.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Database;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\Database;
+
+/**
+ * Get table data if it exists in the database.
+ */
+class ShowTableInfo extends BaseCommand
+{
+    /**
+     * The group the command is lumped under
+     * when listing commands.
+     *
+     * @var string
+     */
+    protected $group = 'Database';
+
+    /**
+     * The Command's name
+     *
+     * @var string
+     */
+    protected $name = 'db:table';
+
+    /**
+     * the Command's short description
+     *
+     * @var string
+     */
+    protected $description = 'Retrieves the selected table information.';
+
+    /**
+     * the Command's usage
+     *
+     * @var string
+     */
+    protected $usage = 'db:table <table_name> [options]';
+
+    /**
+     * The Command's arguments
+     *
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'table_name' => 'The table name for show information.',
+    ];
+
+    /**
+     * The Command's options
+     *
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--show' => 'Retrieves list the names of all database tables.',
+    ];
+
+    /**
+     * get table info
+     */
+    public function run(array $params)
+    {
+        // Connect to database
+        $db        = Database::connect();
+        $getTables = $db->listTables();
+        // The database does not have a table
+        if ($getTables === []) {
+            return CLI::error('Database has no tables!', 'light_gray', 'red');
+        }
+
+        // show all tables name
+        if (CLI::getOption('show')) {
+            CLI::write('list the names of all database tables : ', 'black', 'yellow');
+            CLI::write(implode(' , ', $getTables), 'black', 'blue');
+
+            return CLI::newLine();
+        }
```

Comment:
```suggestion
        if (CLI::getOption('show')) {
            CLI::write("The following is a list of the names of all database tables:\n", 'black', 'yellow');
            CLI::write(implode(', ', $tables), 'black', 'blue');
            CLI::newLine();

            return;
        }
```

Also, I'm not sure if this was considered but the number of tables a DB can have may be so many that the terminal width may be not enough. We may need to wrap the listing.

Also, I'm not sure on the color choice. Black foreground on a yellow background and black foreground on a blue background placed together seems arbitrary to me (but I haven't run this code yet so I'm not yet sure if the color fits well.)

Reply:
@kenjis  and @paulbalandan  Because this part was **worrying for you**. I changed in general.
The output will eventually be as follows.
Please let me know if you see a need to change this.
```
php spark db:table --show

CodeIgniter v4.1.9 Command Line Tool - Server Time: 2022-04-19 05:15:17 UTC-05:00

list the names of all database tables : 
+-----+--------------------------+--------------+-------------+
| #ID | table name               | field number | rows number |
+-----+--------------------------+--------------+-------------+
| 1   | alert_messages           | 11           | 12          |
| 11  | departments              | 8            | 3           |
| 12  | migrations               | 7            | 5           |
| 13  | tickets                  | 12           | 5           |
| 14  | urgencies                | 8            | 6           |
+-----+--------------------------+--------------+-------------+

```